packages/version-select/index.test.tsx

```diff
@@ -76,6 +76,33 @@ describe('<VersionSelect />', () => {
     expect(button).toHaveTextContent('v0.4.x')
   })
 
+  it('should display the correct version after browser-back', () => {
+    useRouterMock.mockImplementation(() => {
+      return ({
+        route: '/commands/[[...page]]',
+        pathname: '/commands/[[...page]]',
+        asPath: '/commands',
+        push: mockPush,
+      } as unknown) as Router
+    })
+
+    const { getByRole, getAllByRole } = render(
+      <VersionSelect {...defaultProps} />
+    )
+    const index = 1
+
+    const combobox = getByRole('combobox')
+    userEvent.click(combobox)
+
+    const options = getAllByRole('option')
+    userEvent.click(options[index])
+
+    history.back()
+
+    const button = getByRole('button')
+    expect(button).toHaveTextContent('v0.5.x (latest)')
```

Comment:
What's the value before the back? Given that we're mocking the router I'm not sure what exactly we're asserting here 🤔

Reply:
It would be `"v0.4.x"` — I can add an assertion before to be safe

Reply:
I don't think this is gonna work how we expect here, cause the router is mocked so the URL isn't going to be changing.

Reply:
I think the only way to test this would be to change the return value of the router mock and then force the component to re-render.

In which case the value of this test case is a little dubious as we already have test cases to assert we are getting the version from the URL, and the behavior of `SelectInput` when `value` changes should be tested there.

Reply:
I believe you're right 🤔

Reply:
I have a test that is failing for `select-input 4.0.3` but working on `select-input 4.0.4`

```ts
  it('should display the correct version after browser-back', () => {
    useRouterMock.mockImplementation(() => {
      return {
        asPath: '/commands',
        push: mockPush,
      } as unknown as Router
    })

    const { getByRole, rerender } = render(<VersionSelect {...defaultProps} />)

    expect(getByRole('button')).toHaveTextContent('v0.5.x (latest)')

    // simulate navigating: latest -> v0.4.x
    useRouterMock.mockImplementation(() => {
      return {
        asPath: '/commands/v0.4.x',
        push: mockPush,
      } as unknown as Router
    })
    rerender(<VersionSelect {...defaultProps} />)

    expect(getByRole('button')).toHaveTextContent('v0.4.x')

   // simulate navigating: v0.4.x -> latest
    useRouterMock.mockImplementation(() => {
      return {
        asPath: '/commands/',
        push: mockPush,
      } as unknown as Router
    })
    rerender(<VersionSelect {...defaultProps} />)

    expect(getByRole('button')).toHaveTextContent('v0.5.x (latest)')
  })
```